### PR TITLE
Fix clippy warnings with Rust 1.72.0

### PR DESCRIPTION
**Pull request by cakebaker**

### CODE REVIEW

**Comment by tertsdiepraam** [comment]
I don't really like the changes from `foo.iter()` to `&*foo`. I suppose just `&` doesn't work? I guess it's fine either way, neither option is _great_.

File: src/uu/factor/src/factor.rs [hunk 1]
```diff
@@ -96,7 +96,7 @@ impl fmt::Display for Factors {
         v.sort_unstable();
 
         let include_exponents = f.alternate();
-        for (p, exp) in v.iter() {
+        for (p, exp) in &*v {
```
Comment (tertsdiepraam) [inline src/uu/factor/src/factor.rs:99]:
```suggestion
        for (p, exp) in v {
```

File: src/uu/factor/src/factor.rs [hunk 2]
```diff
@@ -292,7 +292,7 @@ impl std::ops::BitXor<Exponent> for Factors {
     fn bitxor(self, rhs: Exponent) -> Self {
         debug_assert_ne!(rhs, 0);
         let mut r = Self::one();
-        for (p, e) in self.0.borrow().0.iter() {
+        for (p, e) in &self.0.borrow().0 {
```
Comment (tertsdiepraam) [inline src/uu/factor/src/factor.rs:295]:
```suggestion
        for (p, e) in self.0.borrow().0 {
```

File: src/uu/mktemp/src/mktemp.rs [hunk 3]
```diff
@@ -451,7 +451,7 @@ pub fn dry_exec(tmpdir: &str, prefix: &str, rand: usize, suffix: &str) -> UResul
     // Randomize.
     let bytes = &mut buf[prefix.len()..prefix.len() + rand];
     rand::thread_rng().fill(bytes);
-    for byte in bytes.iter_mut() {
+    for byte in &mut *bytes {
```
Comment (tertsdiepraam) [inline src/uu/mktemp/src/mktemp.rs:454]:
```suggestion
    for byte in bytes {
```

File: src/uu/factor/src/factor.rs (hunk 1 above)
Comment (sylvestre) [reply src/uu/factor/src/factor.rs:99]: yeah, just like you, i don't like the &* syntax :)
if v works, great! if it doesn't, i would prefer to keep v.iter

Comment (tertsdiepraam) [reply src/uu/factor/src/factor.rs:99]: The changes I suggested should work. But there are a few more cases of `&*` and `&mut *` left that I couldn't simplify. I'll go look for a clippy issue related to this. I'm guessing we aren't the only ones who don't like this.

Comment (tertsdiepraam) [reply src/uu/factor/src/factor.rs:99]: Found something related: https://github.com/rust-lang/rust-clippy/issues/11074

Comment (cakebaker) [reply src/uu/factor/src/factor.rs:99]: Ok, I will simplify where possible and going back to `iter` (and disable the warning) in the other cases.

Comment (tertsdiepraam) [reply src/uu/factor/src/factor.rs:99]: Thanks!

**Pull request merged**